linker/Mono.Linker.Steps/MarkStep.cs

```diff
@@ -483,6 +505,20 @@ void ProcessModule (AssemblyDefinition assembly)
 			}
 		}
 
+		void ProcessLazyAttributes ()
+		{
+			while (_lazyAttributes.Count != 0) {
+				var customAttribute = _lazyAttributes.Dequeue ();
+
+				// If an attribute's module has not been marked after processing all types in all assemblies,
+				// then surely nothing is using this attribute and there is no need to mark it
+				if (!Annotations.IsMarked (customAttribute.AttributeType.Resolve ().Module))
```

Comment:
I wanted to make this change as conservative as possible.  What I'm trying to do is catch the case where a reference to an assembly is preserved simply because that assembly defined & used an assembly or module attribute.

I think how we handle attributes needs to be reviewed and improved.  It might involve different modes, i.e. aggressive vs safe.  I don't know.  Someone needs to sit down and give it some thought.  I may at some point, but it's not a priority at the moment.

Now that you mention it though, maybe I should also check `&& !Annotations.IsMarked(customAttribute.AttributeType)`?  I don't think adding that would hurt, but I don't think it's necessary.  If another assembly was using the attribute, it would cause the attribute to be marked...I think..